### ISSUE TITLE
[FR] Show unit of measurement in Sub Part Dopdown ("Add BOM Item" Modal)

### ISSUE THREAD

**Issue by simonkuehling** [inventree/InvenTree#4330]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

When manually adding a BOM Item to an assembly, it would be super handy to see the unit of measurement of a `Sub Part` after selecting it in the drop-down. For all parts that are not in `pieces`-units like paint (liters? milliliters?), textiles (length? weight? and m/mm/kg/g?), I often need to open the part itself to check which units are defined in there.

### Suggested solution

If the unit would be visible after selecting the `Sup Part`, that would be a huge win In my opinion:

![Bildschirmfoto vom 2023-02-10 16-09-01](https://user-images.githubusercontent.com/296454/218127344-b3b30d67-b44f-42d0-9391-59ec1ea9ea7f.png)

### Describe alternatives you've considered

The only alternative is to open another tab, search for the part and open it to check which unit was defined.

### Do you want to develop this?

- [ ] I want to develop this.

**Comment by matmair** [comment]
@simonkuehling  We could certainly extend the renderer for the select. I think this would make sense as badge on the right side as is `available` right now. What do you think?

**Comment by SchrodingersGat** [comment]
Maybe I'm misunderstanding the question - but isn't that already displayed in your example image?

![image](https://user-images.githubusercontent.com/10080325/218206524-484c6d71-e9c3-4336-a2cf-5671400dde60.png)

**Comment by simonkuehling** [comment]
@SchrodingersGat oh, you are right, bad example ;-) - did not see that when taking screenshot-example for illustration. The `Blue Paint` part here is in stock, and those do have the unit in the badge, true. 
For parts that are out of stock the unit is not shown, though. That's where I stumbled upon this during use.

![Bildschirmfoto vom 2023-02-11 09-34-34](https://user-images.githubusercontent.com/296454/218248977-b891e317-ed1a-4d27-bb41-96ddad0bfce1.png)

@matmair adding a badge to the right would be a clean visual solution, I think. Maybe remove the unit from the stock badge then? So it becomes something like
`Available: 287` `Unit: ml`
and
`No stock` `Unit: ml`

**Comment by matmair** [comment]
@simonkuehling sounds clean. Do you want to dev this or should we open it up for contribution?

**Comment by simonkuehling** [comment]
@matmair I could give it a try at least... Is this the right source for the stock badge?

https://github.com/inventree/InvenTree/blob/75c82f4db41ac1312d5939cb7e7faa8d180b16c8/InvenTree/templates/js/translated/part.js#L577

A change here would probably affect other views/forms etc. around InvenTree as well, like the search results - would that be ok?

**Comment by matmair** [comment]
@simonkuehling I think it should be consistent throughout the interface. Should be the right file.

**Comment by simonkuehling** [comment]
@matmair I have two variations to decide on - either explicit with `Unit: kg` or a little lighter with just the unit itself like `kg` (screenshots of each below). Personally I like the bare second variant because it is less repeating text in the UI (especially in the search results) and generally fewer characters, which uses less screen real estate. Do you agree?

Long version:
![Bildschirmfoto vom 2023-02-12 19-07-01](https://user-images.githubusercontent.com/296454/218329054-2f3addc6-001f-434f-a523-4e8716d4b057.png)

Short version:
![Bildschirmfoto vom 2023-02-12 19-06-33](https://user-images.githubusercontent.com/296454/218329051-175e9631-6bc3-4eb0-a8ca-ead382fd7add.png)

I think this is still quite self explanatory even with `No Stock` like here:
![Bildschirmfoto vom 2023-02-12 19-13-50](https://user-images.githubusercontent.com/296454/218329216-371971c4-ffe8-47c0-9e2f-be4fa21fb720.png)

**Comment by matmair** [comment]
Maybe it makes sense to keep it as-is if there is stock and add the long version (1) if there is no stock. That way it should always be logical what is meant - some unit markings can be a bit ambiguous. @inventree/triage thoughts?

**Comment by SchrodingersGat** [comment]
@simonkuehling I think your latest suggestion here is quite unobtrusive :)

**Comment by matmair** [comment]
Looks nice @simonkuehling; I think that is the way to go

**Comment by simonkuehling** [comment]
Here we go - pull request incoming at https://github.com/inventree/InvenTree/pull/4339

**Comment by simonkuehling** [comment]
closing as completed.

**Comment by janujarerat** [comment]
@simonkuehling Any chance #4339 extends the unit of measurement addition to the New Stock Item modal? It would be super helpful to have it there as well!
![image](https://user-images.githubusercontent.com/54828174/223313850-f36cad56-b52d-4eaf-9d86-34ff31ab1731.png)

**Comment by simonkuehling** [comment]
@janujarerat it should be displayed in all part search drop-downs throughout the interface - if I look at the demo server it's looking like it does in the [New Stock Item] modal as well:

![Bildschirmfoto vom 2023-03-07 08-08-24](https://user-images.githubusercontent.com/296454/223350041-4594178c-b359-4358-833f-759a19a70040.png)

Or is this not the modal you are referring to?

**Comment by janujarerat** [comment]
@simonkuehling Yes, that's the one. Looks great, thanks!
